src/game/ingame_menu.c

```diff
@@ -289,32 +289,50 @@ static s32 is_color_code_valid(char *str, s32 strPos) {
 }
 
 /**
- * Get the exact length of a string of any font in pixels, using the given ASCII and UTF-8 tables.
+ * Get the exact width of a string of any font in pixels, using the given ASCII and UTF-8 tables.
  */
-s32 get_string_length(char *str, struct AsciiCharLUTEntry *asciiLut, struct Utf8LUT *utf8LUT) {
-    s32 length = 0;
+s32 get_string_width(char *str, struct AsciiCharLUTEntry *asciiLut, struct Utf8LUT *utf8LUT) {
+    s32 width = 0;
+    s32 maxWidth = 0;
     s32 strPos = 0;
+    char c;
 
-    s32 checkColorCodes = (asciiLut == main_font_lut); // Only check for @ if the font is the main generic font.
+    s32 isGenericFont = (asciiLut == main_font_lut);
 
     asciiLut = segmented_to_virtual(asciiLut);
 
-    while (str[strPos] != '\0') {
-        if (checkColorCodes && str[strPos] == '@') {
-            if (is_color_code_valid(str, strPos + 1)) {
-                strPos += 7;
+    while ((c = str[strPos]) != '\0') {
+        // Handle color codes and tabs if using generic font
+        if (isGenericFont) {
+            if (c == CHAR_COLOR_CODE) {
+                if (is_color_code_valid(str, strPos + 1)) {
+                    strPos += 7;
+                    continue;
+                }
+            } else if (c == '\t') {
+                width += 4 * SPACE_KERNING(asciiLut);
+                strPos++;
                 continue;
             }
         }
-        if (str[strPos] & 0x80) {
-            length += utf8_lookup(utf8LUT, str, &strPos)->kerning;
+        // Handle newlines
+        if (c == '\n') {
+            if (width > maxWidth) {
+                maxWidth = width;
```

Comment:
While this is _a_ solution, I don't know if it's the ideal one. Basically what this accomplishes is left aligning the text and calculating only the center/right offset.

I don't see why this couldn't be calculated in the main print loop tbh. You just need a global or parameter that describes alignment, and if centered or right-aligned, calculate the width / x offset when processing the start of the active line. Are there any details I'm missing here?